clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -366,8 +383,16 @@ public void close() {
         }
     }
 
+    // visible for testing
+    void invokeCompletedOffsetCommitCallbacks() {
+        for (OffsetCommitCompletion completion : completedOffsetCommits)
+            completion.invoke();
+        completedOffsetCommits.clear();
+    }
+
+    public synchronized void commitOffsetsAsync(final Map<TopicPartition, OffsetAndMetadata> offsets, OffsetCommitCallback callback) {
+        invokeCompletedOffsetCommitCallbacks();
```

Comment:
No great reason, but it seemed like a good opportunity to clear pending commits.